localstack-core/localstack/services/s3/utils.py

```diff
@@ -1052,3 +1055,16 @@ def parse_post_object_tagging_xml(tagging: str) -> Optional[dict]:
 
     except Exception:
         raise MalformedXML()
+
+
+def generate_safe_version_id() -> str:
+    # the safe b64 encoding is inspired by the stdlib base64.urlsafe_b64encode
+    # and also using stdlib secrets.token_urlsafe, but with a different alphabet adapted for S3
+    # VersionId cannot have `-` in it, as it fails in XML
+    tok = token_bytes(24)
```

Comment:
Q: Any specific reason we have hardcoded 24 bytes long version id here? From the documentation: 

> Version IDs are Unicode, UTF-8 encoded, URL-ready, opaque strings that are no more than 1,024 bytes long

it says it can be anything less than 1024 bytes? 🤔

Reply:
Yes, it seems that currently, AWS only return base64 encoded values that are 32 characters long, giving us this 24 bytes secrets. If AWS does change their format, we'll adapt in consequence as the test is pretty strict against AWS as well for the length of the token. I've generated 100s of `VersionId` and they all followed that rule, even if not written anywhere 😅

Reply:
Yes makes sense, I was wondering why they would have written `1024` in the documentation, weird 🫠

Reply:
I think it's what I have said in the PR description: they're pretty vague about the format, probably to have some leeway in doing changes in the future (I think I've seen the length of the VersionId change over time, or maybe I read about it)

Reply:
Thanks a lot for the review! 🙏